File: frontend/src/types/office.ts
export interface Office {
  id: number;
  code: string;
  postalCode: string;
  name: string;
  cityCode: number;
  wardCode: number;
  detail: string;
  latitude: number;
  longitude: number;
  email: string;
  phoneNumber: string;
  openingTime: string;
  closingTime: string;
  type: string;
  //manager: Employee;
  capacity: number;
  notes: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface OfficeSearchRequest {
  page?: number; 
  limit?: number; 
  search?: string; 
  city?: boolean;
  ward?: boolean;
}

export interface AdminOffice {
  id: number;
  code: string;
  postalCode?: string;
  name: string;
  email?: string;
  phoneNumber?: string;
  type?: string;
  status?: string;
  latitude?: number;
  longitude?: number;
  openingTime?: string;
  closingTime?: string;
  capacity?: number;
  notes?: string;
  cityCode?: number;
  wardCode?: number;
  detail?: string;
  createdAt: string;
}

export interface CreateOfficePayload {
  code: string;
  postalCode?: string;
  name: string;
  latitude: number;
  longitude: number;
  email?: string;
  phoneNumber?: string;
  openingTime?: string;
  closingTime?: string;
  type?: string;
  status?: string;
  capacity?: number;
  notes?: string;
  wardCode: number;
  cityCode: number;
  detailAddress: string;
}

export type UpdateOfficePayload = Partial<CreateOfficePayload>;

File: backend/src/main/java/com/logistics/entity/Office.java
package com.logistics.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

import org.hibernate.envers.Audited;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import com.logistics.enums.OfficeStatus;
import com.logistics.enums.OfficeType;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Audited
@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "offices")
@EntityListeners(AuditingEntityListener.class)
public class Office {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, unique = true, length = 50)
    private String code; // Mã nội bộ, phải unique (PO_ID)

    // Thêm này cho in vận đơn
    @Column(length = 10, nullable = true)
    private String postalCode; // Mã bưu chính, không cần unique (PO_Mã tỉnh bưu cục_Mã xã bưu cục) dùng hiển thị trên vận đơn

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false, unique = true)
    private String name; // Tên bưu cục, unique

    // Địa chỉ bưu cục
    @Column(nullable = false)
    private Integer cityCode;
    
    @Column(nullable = false)
    private Integer wardCode;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String detail;

    @Column(nullable = false, precision = 12, scale = 7)
    private BigDecimal latitude; // Vĩ độ

    @Column(nullable = false, precision = 12, scale = 7)
    private BigDecimal longitude; // Kinh độ

    @Column(nullable = false, length = 100)
    private String email;

    @Column(nullable = false, length = 15, unique = true)
    private String phoneNumber; // Số điện thoại liên hệ

    @Column(nullable = false)
    private LocalTime openingTime = LocalTime.of(7, 0); // Giờ mở cửa

    @Column(nullable = false)
    private LocalTime closingTime = LocalTime.of(17, 0); // Giờ đóng cửa

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OfficeType type = OfficeType.POST_OFFICE;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OfficeStatus status = OfficeStatus.ACTIVE;

    // Quan hệ với nhân viên
    @OneToMany(mappedBy = "office", cascade = CascadeType.ALL)
    private List<Employee> employees;

    // Quản lý riêng
    @OneToOne
    @JoinColumn(name = "manager_id", nullable = true)
    private Employee manager;

    // Sức chứa tối đa (số kiện hàng tối đa của bưu cục)
    @Column
    private Integer capacity;

    // Ghi chú thêm
    @Lob
    private String notes;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}